src/hooks/tests/use-fullscreen.test.ts

```diff
@@ -0,0 +1,78 @@
+import { renderHook } from '@testing-library/react';
+import { act } from 'react';
+import { getIpcApi } from '../../lib/get-ipc-api';
+import { useFullscreen } from '../use-fullscreen';
+import { useIpcListener } from '../use-ipc-listener';
+
+jest.mock( '../../lib/get-ipc-api' );
+jest.mock( '../use-ipc-listener' );
+
+const mockIpcApi = {
+	isFullscreen: jest.fn(),
+};
+
+( getIpcApi as jest.Mock ).mockReturnValue( mockIpcApi );
+
+describe( 'useFullscreen', () => {
+	beforeEach( () => {
+		jest.clearAllMocks();
+	} );
+
+	it( 'should initialize with false and update when isFullscreen resolves', async () => {
+		mockIpcApi.isFullscreen.mockResolvedValue( true );
+		const { result } = renderHook( () => useFullscreen() );
+
+		expect( result.current ).toBe( false );
+
+		await act( async () => {
+			await Promise.resolve();
+		} );
```

Comment:
```suggestion
		await waitFor( () => {
			expect( mockIpcApi.isFullscreen ).toHaveBeenCalledTimes( 1 );
		} );
```
Optional, but I found the current code a bit cryptic. I think this clarifies the intention.

Reply:
Much cleaner approach, thank you!